### PR TITLE
Informal writeup of the proof that group homomorphisms are epi iff surjective

### PR DESCRIPTION
**Pull request by EgbertRijke**
This PR only adds an informal writeup of a proof that a group homomorphism is surjective iff it is an epi. A proof without excluded middle can be given using the theory of acyclic maps, which I'm still working on with Ulrik and Tom. I thought that the writeup could still be worthwhile to have on our website.

### CODE REVIEW

File: src/group-theory/epimorphisms-groups.lagda.md [hunk 1]
```diff
@@ -61,3 +68,42 @@ module _
   is-epi-iso-Group =
     is-epi-iso-Large-Precategory Group-Large-Precategory l3 G H f
 ```
+
+### A group homomorphism is surjective if and only if it is an epimorphism
+
+**Proof using the law of excluded middle:** The forward direction of this claim
+is the easier of the two directions, and this part of the proof doesn't require
+the [law of exculded middle](foundation.law-of-excluded-middle.md). If `f` is
```
Comment (fredrik-bakke) [inline src/group-theory/epimorphisms-groups.lagda.md:76]:
```suggestion
the [law of excluded middle](foundation.law-of-excluded-middle.md). If `f` is
```

File: src/group-theory/epimorphisms-groups.lagda.md [hunk 2]
```diff
@@ -61,3 +68,42 @@ module _
...
+[surjective](foundation.surjective-maps.md) and `g h : H → K` are two group
+homomorphisms such that `g ∘ f ＝ h ∘ f`, then to show that `g ＝ h` it suffices
+to show that `g y ＝ h y` for any `y : H`. Since we are proving a
+[proposition](foundation.propositions.md) and `f` is assumed to be surjective,
+we may assume `x : G` equipped with an
+[identification](foundation.identity-types.md) `f x ＝ y`. It therefore suffices
+to show that `g (f x) ＝ h (f x)`, which was assumed.
+
+For the converse, suppose that `f : G → H` is an epimorphism and consider the
+subgroup `I := im f` of `H`. We first show that `I` is
```
Comment (fredrik-bakke) [inline src/group-theory/epimorphisms-groups.lagda.md:86]:
```suggestion
[image subgroup](group-theory.images-of-group-homomorphisms.md) `I := im f` of `H`. We first show that `I` is
```

File: src/group-theory/epimorphisms-groups.lagda.md [hunk 3]
```diff
@@ -61,3 +68,42 @@ module _
...
+[normal](group-theory.normal-subgroups.md), and then we show that `I ＝ H`.
+
+In order to show that `I` is normal, we want to show that `I` has only one
+conjugacy class, namely itself. Consider the group `K` of permutations of the
+set of [conjugate](group-theory.conjugation.md) subgroups of the subgroup `I` of
```
Comment (fredrik-bakke) [inline src/group-theory/epimorphisms-groups.lagda.md:91]:
```suggestion
set of [conjugate](group-theory.conjugation.md) [subgroups](group-theory.subgroups.md) of the subgroup `I` of
```

**Review by fredrik-bakke — APPROVED** [review]

**Comment by fredrik-bakke** [comment]
Very nice! I like the idea of including write-ups like this. Even if the proof is formalized, having an informal argument as well is helpful for multiple reasons.
- It gives an overview of the formal proof where the writer can selectively omit less important details, and add details for context.
- It's a resource for readers who are learning to read agda-unimath.
- Even if the reader is not interested in learning agda-unimath, the informal argument can help them understand some of the formalization work.

**Comment by EgbertRijke** [comment]
A fourth reason might be that if someone wants to formalize something that already has a writeup, they already have a suggested proof that they can follow.

**Comment by EgbertRijke** [comment]
I noticed that most of our PRs are now co-authored between us, because of how `github` suggestion work. It reminds me of how all the Beatles songs are written by Lennon-McCartny

**Pull request merged**